include/psa/crypto.h

```diff
@@ -203,30 +206,51 @@ psa_status_t psa_crypto_init(void);
  */
 typedef struct psa_key_attributes_s psa_key_attributes_t;
 
-/** Declare a key as persistent.
+/** Declare a key as persistent and set its key identifier.
+ *
+ * If the attribute structure declares the key as volatile (which is
+ * the default content of an attribute structure), this function sets
+ * the lifetime attribute to #PSA_KEY_LIFETIME_PERSISTENT.
  *
  * This function does not access storage, it merely fills the attribute
- * structure with given values. The persistent key will be written to
+ * structure with given value. The persistent key will be written to
  * storage when the attribute structure is passed to a key creation
  * function such as psa_import_key(), psa_generate_random_key(),
  * psa_generate_derived_key() or psa_copy_key().
  *
- * This function overwrites any identifier and lifetime values
- * previously set in \p attributes.
- *
  * This function may be declared as `static` (i.e. without external
  * linkage). This function may be provided as a function-like macro,
  * but in this case it must evaluate each of its arguments exactly once.
  *
  * \param[out] attributes       The attribute structure to write to.
  * \param id                    The persistent identifier for the key.
+ */
+static void psa_set_key_id(psa_key_attributes_t *attributes,
+                           psa_key_id_t id);
```

Comment:
What should happen if the caller specifies `id` 0 here? Is this invalid? I think we need some more documentation here.

Reply:
The functions have special cases to preserve the invariant that `id = 0` if `lifetime = VOLATILE`. But `id = 0` with `lifetime ≠ VOLATILE` is ok as the intermediate state if you call `set_key_lifetime` first, then `set_key_id`. Do you think there should be a special case for `id=0`?

Reply:
`id = 0` should be allowed as an intermediate state, and when the attributes are actually used, an error can be returned if the id is invalid.